### ISSUE TITLE
[Meta] Top Sites

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#6190]
### User Stories
- [x] #6757 As a user, I want to easily access my top visited sites on my homepage, so I can get to the information I need faster.
- [x] #6758 As a user, I want to be able to explicitly add one of my favorite sites to the home screen
- [x] #6759 As a user, I would like to be able to remove a top site I no longer want to see
- [x] #8126 Default Top Sites 

See [feature brief](https://docs.google.com/document/d/1UyBj9c1CXktL6FRZeHd1WzBRxQLlwB0WBZUi-5DwnoE/edit#) for more details

### Dependencies
- [x] #6602 Optimize homescreen should be done first
~~(if choose to go the auto-top sites route) Add AS query for most frequently visited sites https://github.com/mozilla/application-services/issues/2163~~
- [x] Migration piece to get pinned sites https://github.com/mozilla-mobile/android-components/issues/5126
- [x] Create a Top Site Storage component  https://github.com/mozilla-mobile/android-components/issues/5315

### Acceptance criteria

- [x] I see Pocket Top Articles on the homescreen
- [x] I see default Top Sites on the homescreen
- [x] I can add a top site
- [x] I can remove a top site
- [x] I can open a top sites in regular browsing mode
- [x] I can open a top sites in private mode
- [x] [Telemetry] Top Sites #7525
- [x] [Telemetry] Default top sites #8125

### UX flow
https://share.goabstract.com/f9ae2c95-4a95-4b0e-917c-d87b9854a600?mode=design&sha=2fac94ac7fd2676633043e1967dc696c5071a8da

**Comment by vesta0** [comment]
A-S ticket filed [here](https://github.com/mozilla/application-services/issues/2163).

**Comment by liuche** [comment]
Next steps here: we'll break out into a user-story filing (@boek and @topotropic)

**Comment by liuche** [comment]
Chatted with @linacambridge and @grigoryk about what it would take to pull frequent sites from history (AS query), and pinned sites (from migration) (and updated comment 0).

If we want to generate a static top sites, once, and then use that everywhere later, the complexity probably is in knowing "when to pull from top/pinned sites" based on a combination of migration, signing into fxa, starting to sync a history engine.

**Comment by Cheap-Skate** [comment]
Minor comment re the UX flow in Comment 0

It's not clear to me what "add to Firefox home" means i.e. I don't really recognise the current "tabs tray" as a home page.  I think I (and perhaps others) have the opposed view to some of you Mozilla guys, I live my life in tabs and think of the tabs tray as merely a route to a new tab, not a destination or home page. I'm sure we'll change our views in time but I think you might use clearer wording for us Luddites....

So my simple suggestion would be change the wording away from "Add to Firefox home" for example re-use the plus sign and "pin site" wording that's used on the Home screen itself

e.g.

On the Home screen "Pin site +"

On the 3 dot menu "+ Pin site on Home screen"
or   "+ Add to Top Sites"
or ... something   ??

**Comment by Cheap-Skate** [comment]
Is there a plan to have Fenix auto-add frequently visited sites to Top Sites?

I think that's really important, I know casual users who rely on the browser identifying and presenting Top Sites i.e. they don't bother pinning them themselves, they just use what the browser suggests.

**Comment by Cheap-Skate** [comment]
Final comment:- I really like the idea that I can have as many rows of Top Sites as I like!

(Which makes me think.... folders on the Top Sites page? Tap a folder to open it, see Top Sites inside? Then it replicates the Android Home screen where I can place a folder on the Home screen and put apps in it)  Mmmmmm

**Comment by cadeyrn** [comment]
> Is there a plan to have Fenix auto-add frequently visited sites to Top Sites?

It's the first checkbox in comment 0:

> #6757 As a user, I want to easily access my top visited sites on my homepage, so I can get to the information I need faster.

This one is not about manually adding top sites because the next one is about manually adding sites:

>  #6758 As a user, I want to be able to explicitly add one of my favorite sites to the home screen

**Comment by madb1lly** [comment]
First time I've seen that UX mobkup. I agree with @Cheap-Skate that I wouldn't necessarily know what "Firefox Home" means, coming from Fennec. I personally struggle to see what is wrong with the Fennec Awesome-page, that hasn't been well explained yet AFAICT.

And what on earth is a snackbar?! :D

Cheers 🙂

**Comment by cadeyrn** [comment]
> And what on earth is a snackbar?! :D

https://material.io/components/snackbars/

**Comment by djc** [comment]
> It's not clear to me what "add to Firefox home" means i.e. I don't really recognise the current "tabs tray" as a home page. I think I (and perhaps others) have the opposed view to some of you Mozilla guys, I live my life in tabs and think of the tabs tray as merely a route to a new tab, not a destination or home page. I'm sure we'll change our views in time but I think you might use clearer wording for us Luddites....

Interestingly, this is exactly what I wrote up in #2484, but it appears this hasn't really gotten much attention.

**Comment by st3fan** [comment]
Just adding a note here that the A-C work in _Create a Top Site Storage Component_  https://github.com/mozilla-mobile/android-components/issues/5315 was merged through https://github.com/mozilla-mobile/android-components/pull/5360 - if this does not meet the requirements for this feature, please file issues in A-C and mark them to be included in our sprint planning.

**Comment by vesta0** [comment]
@djc @Cheap-Skate FYI we are also making changes to the tab tray and the back button that aligns better with this approach. 

@topotropic I think it may be a good idea to use something like "pin site to home screen" instead of "add to Firefox home" what do you think?

**Comment by djc** [comment]
@vesta0 sounds good! Are those changes being tracked in some issue or Bugzilla bug that I could follow?

**Comment by topotropic** [comment]
> @topotropic I think it may be a good idea to use something like "pin site to home screen" instead of "add to Firefox home" what do you think?

I find "home screen" problematic, because we also use it to refer to the device's home screen ("Add to Home screen" in the menu). "Firefox new tab" or "Firefox new window" could be an alternative. I originally choose the "Firefox home" terminology because we also use it on Firefox on desktop.  

@mheubusch can you chime in here? Thanks!

**Comment by Cheap-Skate** [comment]
Yeah on Desktop it's clear, the app (Firefox) has a Home page; the OS (Windows/Mac/Linux) has no Home screen, typically it's a Desktop.

On Android I think many users understand that Home is the Android Home screen/launcher.   So what is Fenix's Top Sites & tab switcher page called? IMHO  "Home" is too confusing. How about  "Firefox Central" :)

**Comment by madb1lly** [comment]
In Fennec the new tab page _is_ called "Homepage" in settings, but in terms of pinning sites it's not called that AFAICT. I think the new page is unofficially called the "Awesome Page" isn't it? Similar to how the combined search and address bar is the "Awesome Bar".

Slightly off-topic, but can someone please give me the reference/link to the discussion/explanation of why the new tab page from Fennec isn't being retained for use in Fenix, and why in Fenix we are trying to cram tabs _and_ new tab page stuff into a single screen?

Cheers :)

**Comment by Cheap-Skate** [comment]
> Slightly off-topic, but can someone please give me the reference/link to the discussion/explanation of why the new tab page from Fennec isn't being retained for use in Fenix, and why in Fenix we are trying to cram tabs _and_ new tab page stuff into a single screen?

I think this has been the plan from Day 1, idea is to make Fenix Home screen centric. I dislike it, I want it to be tab centric like Fennec. I filed #570 now closed. There are various howls of dismay at the loss of the Fennec Awesomescreen (some from me) but Mozilla wants to try something new, fair enough. There is a redesign of the Home screen coming #1063 and #6602 and https://mozilla.invisionapp.com/share/7PSR7KCSW6Z#/screens/372749649_Latest. But it's still different to what we're used to. Somewhere I read that the Awesomescreen was not well liked by (casual?) users.

I am currently concerned that tapping a Top Site will always open it in a new tab which I think will be a PITA, my morning skim of a dozen news sites will result in an overload of tabs all of which will need to be killed or reloaded the next time I skim. But we shall see.

**Comment by mheubusch** [comment]
@topotropic Your explanation as to why "Firefox home" is correct. We are avoiding confusion with Android's home screen. I think if we observe that users can't complete the task or don't discover it at all because of the label we should consider a new phrase and test that for task completion and discovery before rolling it out.

**Comment by yoasif** [comment]
I saw a post about this on reddit today and I found it really hard to find the feature. There are no top sites by default, so even though I knew the feature existed, I actually went to the build info to see if my build had it.

**Comment by andreicristianpetcu** [comment]
I am tracking this feature and I also found out from Reddit :)

**Comment by sblatz** [comment]
Can this meta be closed? :)

**Comment by vesta0** [comment]
@sblatz I am just waiting for the telemetry ticket #8125 to be closed.

**Comment by djc** [comment]
I'm confused. I understood Top Sites to be about a list of sites collected from frecency data from the user's profile. At least, that's what I get on desktop Firefox and what I used to use in Fennec. As far as I see, currently there's only Pocket/Wikipedia/YouTube. How could this issue be closed?

**Comment by Dunexus** [comment]
Also, they are not synced with desktop top sites.

**Comment by gabrielluong** [comment]
> I'm confused. I understood Top Sites to be about a list of sites collected from frecency data from the user's profile. At least, that's what I get on desktop Firefox and what I used to use in Fennec. As far as I see, currently there's only Pocket/Wikipedia/YouTube. How could this issue be closed?

> Also, they are not synced with desktop top sites.

This is totally understandable. These items are on my list of things to do. We don't necessarily have to cover these features in this meta since we can just as easily refer to these features with the Feature:TopSites label.

**Comment by vesta0** [comment]
Is there an update on this bug?

**Comment by vesta0** [comment]
All the issues sited here are now closed :)

**Comment by madb1lly** [comment]
Hi @vesta0,

By "My top sites" I had understood my most frequently visited sites, just like in Fennec and Desktop. Isn't this part of this scope of this meta issue? Or is there another issue for this?

Thanks 🙂

**Comment by boek** [comment]
Hey @madb1lly,

Yes that feature is being tracked here: https://github.com/mozilla-mobile/fenix/issues/10342
